mrp_mto_with_stock/tests/test_mrp_mto_with_stock.py

```diff
@@ -68,18 +67,11 @@ def test_manufacture_with_forecast_stock(self):
 
         self.assertEqual(self.production.availability, 'partially_available')
 
-        self.assertEquals(self.subproduct1.virtual_available, 0)
+        self.assertEquals(self.subproduct1.virtual_available, -3)
```

Comment:
Well, I didn't check in deep but my first thought here is that this is not the way to migrate tests. If tests are failing change the logic not adapt the result to what you are getting.

Reply:
@lreficent Here is explanation for -3 virtual quantity. 

* Update `self.subproduct1` with `2` quantity and MO is generated with `5` quantity. So `2-5=-3`

I will do more analysis on it and update you.

Reply:
Yes, of course but the module should use that 2 units and then create a procurement for the remaining 3. The previous 0 was there for some reason, that's what I meant.

Reply:
I would recommend you to review the functionality in v10 and understand it a little bit more and then come back to v11 so you will know what to expect and could figure out what to fix

Reply:
@bodedra In v10, the method `action_assign` in https://github.com/OCA/manufacture/blob/10.0/mrp_mto_with_stock/models/mrp_production.py#L20 is supposed to create a new procurement order to fulfill the missing quantity of the component.

In v11, you need to migrate the method to create a new purchase order or manufacturing order (depending on the product configuration). This will increase the forecast quantity and it will match the expected quantity set in the tests.

Reply:
Not sure with the scope of this module, looking at the code I haven't seen regarding orderpoints on the procurement should we also need to consider it?

Reply:
If an MO that triggered for a PO is cancelled what would happen to the PO? What if PO is on Draft/RFQ state? How about if PO is on confirmed state?

Reply:
@lreficent @max3903 @agyamuta 

I still have problem in following code.

    qty_to_procure = move.remaining_qty - move.reserved_availability

Request to provide few guidance on it.

I have improved logic and test case. Would you please provide me feedback on it.